sql/opt/exec/execbuilder: TestExecBuild failed

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2604753&tab=buildLog) on [master@baf34b241c9ee27530f0a3a988149bfd7cca3f6d](https://github.com/cockroachdb/cockroach/commits/baf34b241c9ee27530f0a3a988149bfd7cca3f6d):

```
    logic.go:2528: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1881: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1445: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1447: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1372: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:481: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:631: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210122 06:12:20.610674 102683 sql/logictest/logic.go:3519  
pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
    logic.go:2261: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210122 06:12:20.611785 102683 sql/logictest/logic.go:3544  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3087: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3087: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
E210122 06:12:20.615009 103175 sql/stats/automatic_stats.go:302  failed to refresh stats: node unavailable; try another peer
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (1.07s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (239.76s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2607612&tab=buildLog) on [master@7a5d3d0f9465c4d6a6f204ac6a672a027cf585a6](https://github.com/cockroachdb/cockroach/commits/7a5d3d0f9465c4d6a6f204ac6a672a027cf585a6):

```
NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1883: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1445: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1447: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1372: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:481: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:631: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210123 06:11:34.586149 101448 sql/logictest/logic.go:3523  
pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210123 06:11:34.587197 101448 sql/logictest/logic.go:3548  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (0.98s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (211.38s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2609829&tab=buildLog) on [master@0455055fafcef7c028b4af8e4efe242b89e7f8a3](https://github.com/cockroachdb/cockroach/commits/0455055fafcef7c028b4af8e4efe242b89e7f8a3):

```
NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1883: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1446: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1448: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1373: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:482: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:631: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210124 06:12:10.789047 102167 sql/logictest/logic.go:3523  
pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210124 06:12:10.790187 102167 sql/logictest/logic.go:3548  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (1.08s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (227.62s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2611757&tab=buildLog) on [master@48fbbe7313fc70f889fa66c5749914f57ddd5a91](https://github.com/cockroachdb/cockroach/commits/48fbbe7313fc70f889fa66c5749914f57ddd5a91):

```
runtime/asm_amd64.s:1374: goexit()

NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1883: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1446: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1448: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1373: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:482: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:631: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210125 06:11:07.617453 103271 sql/logictest/logic.go:3523  
pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210125 06:11:07.618662 103271 sql/logictest/logic.go:3548  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (1.13s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2615398&tab=buildLog) on [master@d86781c07065421f4a4d8bf5d988900ab07fdce5](https://github.com/cockroachdb/cockroach/commits/d86781c07065421f4a4d8bf5d988900ab07fdce5):

```
E210126 06:11:27.824692 101912 sql/logictest/logic.go:3536  [-] 4 +NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1891: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1446: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1448: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1373: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:482: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:631: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210126 06:11:27.829924 101912 sql/logictest/logic.go:3523  [-] 5  
E210126 06:11:27.829924 101912 sql/logictest/logic.go:3523  [-] 5 +pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,5,6,14), got (1,2,5,6)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210126 06:11:27.831003 101912 sql/logictest/logic.go:3548  [-] 6  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (1.01s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (218.02s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2618885&tab=buildLog) on [master@f7c5898f3d552f7ab0751cdd9ffa95cdfd6b8a76](https://github.com/cockroachdb/cockroach/commits/f7c5898f3d552f7ab0751cdd9ffa95cdfd6b8a76):

```
E210127 06:12:15.773843 102599 sql/logictest/logic.go:3536  [-] 7 +NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1891: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1446: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1448: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1373: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:482: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:631: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210127 06:12:15.777568 102599 sql/logictest/logic.go:3523  [-] 8  
E210127 06:12:15.777568 102599 sql/logictest/logic.go:3523  [-] 8 +pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210127 06:12:15.778722 102599 sql/logictest/logic.go:3548  [-] 9  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (0.99s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (227.75s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2622649&tab=buildLog) on [master@1a459a81dba35b6a091f0a2954aa33d50f1e5d24](https://github.com/cockroachdb/cockroach/commits/1a459a81dba35b6a091f0a2954aa33d50f1e5d24):

```
E210128 06:11:48.764031 101541 sql/logictest/logic.go:3536  [-] 5 +
E210128 06:11:48.764031 101541 sql/logictest/logic.go:3536  [-] 5 +NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1891: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1446: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1448: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1373: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:482: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:627: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210128 06:11:48.767729 101541 sql/logictest/logic.go:3523  [-] 6  
E210128 06:11:48.767729 101541 sql/logictest/logic.go:3523  [-] 6 +pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210128 06:11:48.768722 101541 sql/logictest/logic.go:3548  [-] 7  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
E210128 06:11:48.774204 102027 sql/stats/automatic_stats.go:302  [-] 8  failed to refresh stats: node unavailable; try another peer
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (1.01s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2626249&tab=buildLog) on [master@86c619f768a4d6f4cc80efb5c9479e5fc0fd72c1](https://github.com/cockroachdb/cockroach/commits/86c619f768a4d6f4cc80efb5c9479e5fc0fd72c1):

```
E210129 06:11:38.915804 101127 sql/logictest/logic.go:3536  [-] 3 +NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1886: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1451: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1453: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1378: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:483: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:627: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210129 06:11:38.919411 101127 sql/logictest/logic.go:3523  [-] 4  
E210129 06:11:38.919411 101127 sql/logictest/logic.go:3523  [-] 4 +pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210129 06:11:38.920365 101127 sql/logictest/logic.go:3548  [-] 5  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (0.97s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (207.33s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2629403&tab=buildLog) on [master@171cc86d6d84c110197485b5d886d77cddc12399](https://github.com/cockroachdb/cockroach/commits/171cc86d6d84c110197485b5d886d77cddc12399):

```
E210130 06:12:41.914584 101617 sql/logictest/logic.go:3536  [-] 5 +NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1886: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1451: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1453: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1378: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:483: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:627: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210130 06:12:41.918680 101617 sql/logictest/logic.go:3523  [-] 6  
E210130 06:12:41.918680 101617 sql/logictest/logic.go:3523  [-] 6 +pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210130 06:12:41.919636 101617 sql/logictest/logic.go:3548  [-] 7  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (1.04s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (218.69s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2631443&tab=buildLog) on [master@e4ac407b14ed315662050f09670707cdf8ad9bfe](https://github.com/cockroachdb/cockroach/commits/e4ac407b14ed315662050f09670707cdf8ad9bfe):

```
E210201 06:12:03.264306 101355 sql/logictest/logic.go:3536  [-] 5 +NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:30: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1886: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1451: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1453: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1378: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:483: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:627: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210201 06:12:03.267951 101355 sql/logictest/logic.go:3523  [-] 6  
E210201 06:12:03.267951 101355 sql/logictest/logic.go:3523  [-] 6 +pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
--- done: testdata/inverted_join_multi_column with config local: 5 tests, 2 failures
E210201 06:12:03.268970 101355 sql/logictest/logic.go:3548  [-] 7  testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:57: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:57: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (1.02s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (213.28s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2635613&tab=buildLog) on [master@0e6727832d58faf0f900601cd6fa6807e0a2ba75](https://github.com/cockroachdb/cockroach/commits/0e6727832d58faf0f900601cd6fa6807e0a2ba75):

```
E210202 06:12:25.128923 101797 sql/logictest/logic.go:3536  [-] 6 +NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:27: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1885: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:910: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:796: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:641: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:119: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1450: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1452: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1377: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:484: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:627: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210202 06:12:25.134170 101797 sql/logictest/logic.go:3523  [-] 7  
E210202 06:12:25.134170 101797 sql/logictest/logic.go:3523  [-] 7 +pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
--- done: testdata/inverted_join_multi_column with config local: 4 tests, 2 failures
E210202 06:12:25.135259 101797 sql/logictest/logic.go:3548  [-] 8  testdata/inverted_join_multi_column:54: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:54: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:54: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (1.07s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (222.53s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2639234&tab=buildLog) on [master@4736b8dd2c36b58fffbb0a665690e2c0e719d7c3](https://github.com/cockroachdb/cockroach/commits/4736b8dd2c36b58fffbb0a665690e2c0e719d7c3):

```
E210203 06:11:47.091528 101354 sql/logictest/logic.go:3536  [-] 4 +runtime/asm_amd64.s:1374: goexit()
E210203 06:11:47.091528 101354 sql/logictest/logic.go:3536  [-] 4 +
E210203 06:11:47.091528 101354 sql/logictest/logic.go:3536  [-] 4 +NOTE: internal errors may have more details in logs. Use -show-logs.
    logic.go:2532: 
        
        testdata/inverted_join_multi_column:27: EXPLAIN SELECT j1.k
        FROM j1 INNER INVERTED JOIN j2
        ON i IN (10, 20) AND j2.j @> j1.j
        expected success, but found
        (XX000) internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
        relational.go:341: in buildRelational()
        DETAIL: stack trace:
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:341: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1448: buildLookupJoin()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:218: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:683: buildProject()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:199: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:127: func1()
        github.com/cockroachdb/cockroach/pkg/sql/opt_exec_factory.go:1885: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain/explain_factory.og.go:880: ConstructExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/statement.go:120: buildExplain()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:275: buildRelational()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:173: build()
        github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:135: Build()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:565: runExecBuilder()
        github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:261: makeOptimizerPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:918: makeExecPlan()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:804: dispatchToExecutionEngine()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:642: execStmtInOpenState()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:120: execStmt()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1462: func1()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1464: execCmd()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1389: run()
        github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:489: ServeConn()
        github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:627: func1()
        runtime/asm_amd64.s:1374: goexit()
        
        NOTE: internal errors may have more details in logs. Use -show-logs.
E210203 06:11:47.095768 101354 sql/logictest/logic.go:3523  [-] 5  
E210203 06:11:47.095768 101354 sql/logictest/logic.go:3523  [-] 5 +pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
    logic.go:2265: 
         pq: internal error: exec columns do not match opt columns: expected (1,2,14-16), got (1,2,15,16)
--- done: testdata/inverted_join_multi_column with config local: 4 tests, 2 failures
E210203 06:11:47.096846 101354 sql/logictest/logic.go:3548  [-] 6  testdata/inverted_join_multi_column:54: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/inverted_join_multi_column:54: error while processing
    logic.go:3091: testdata/inverted_join_multi_column:54: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/local/inverted_join_multi_column (1.04s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/opt/exec/execbuilder/relational.go
+++ b/pkg/sql/opt/exec/execbuilder/relational.go
@@ -332,18 +332,14 @@ func (b *Builder) buildRelational(e memo.RelExpr) (execPlan, error) {
 	// In test builds, assert that the exec plan output columns match the opt
 	// plan output columns.
 	if util.CrdbTestBuild {
-		// TODO(sumeer): paired inverted joins produce extra columns unknown to the
-		// optimizer.
-		if e.Op() != opt.InvertedJoinOp {
-			optCols := e.Relational().OutputCols
-			var execCols opt.ColSet
-			ep.outputCols.ForEach(func(key, val int) {
-				execCols.Add(opt.ColumnID(key))
-			})
-			if !execCols.Equals(optCols) {
-				return execPlan{}, errors.AssertionFailedf(
-					"exec columns do not match opt columns: expected %v, got %v. op: %T", optCols, execCols, e)
-			}
+		optCols := e.Relational().OutputCols
+		var execCols opt.ColSet
+		ep.outputCols.ForEach(func(key, val int) {
+			execCols.Add(opt.ColumnID(key))
+		})
+		if !execCols.Equals(optCols) {
+			return execPlan{}, errors.AssertionFailedf(
+				"exec columns do not match opt columns: expected %v, got %v. op: %T", optCols, execCols, e)
 		}
 	}
 

--- a/pkg/sql/opt/memo/logical_props_builder.go
+++ b/pkg/sql/opt/memo/logical_props_builder.go
@@ -2171,6 +2171,7 @@ func (h *joinPropsHelper) outputCols() opt.ColSet {
 	//
 	//   1. semi and anti joins, which only project the left columns
 	//   2. lookup joins, which can project a subset of input columns
+	//   3. inverted joins, which can project a subset of input columns
 	//
 	var cols opt.ColSet
 	switch h.joinType {
@@ -2184,6 +2185,10 @@ func (h *joinPropsHelper) outputCols() opt.ColSet {
 	if lookup, ok := h.join.(*LookupJoinExpr); ok {
 		// Remove any columns that are not projected by the lookup join.
 		cols.IntersectionWith(lookup.Cols)
+	}
+	if inv, ok := h.join.(*InvertedJoinExpr); ok {
+		// Remove any columns that are not projected by the inverted join.
+		cols.IntersectionWith(inv.Cols)
 	}
 
 	return cols

--- a/pkg/sql/opt/memo/testdata/stats/inverted-join
+++ b/pkg/sql/opt/memo/testdata/stats/inverted-join
@@ -270,12 +270,12 @@ project
       ├── stats: [rows=3]
       ├── fd: (1)-->(2,3)
       ├── inner-join (inverted json_multi_col@sj_idx [as=t1])
-      │    ├── columns: t2.j:8(jsonb) "inverted_join_const_col_@3":11(string!null) t1.k:12(int!null) s:14(string!null)
+      │    ├── columns: t2.j:8(jsonb) t1.k:12(int!null) s:14(string!null)
       │    ├── prefix key columns: [11] = [14]
       │    ├── inverted-expr
       │    │    └── t1.j:13 @> t2.j:8 [type=bool]
       │    ├── stats: [rows=3, distinct(11)=3, null(11)=0, distinct(12)=2.99565406, null(12)=0, distinct(14)=3, null(14)=0]
-      │    ├── fd: (12)-->(14), (11)==(14), (14)==(11)
+      │    ├── fd: (12)-->(14)
       │    ├── inner-join (cross)
       │    │    ├── columns: t2.j:8(jsonb) "inverted_join_const_col_@3":11(string!null)
       │    │    ├── multiplicity: left-rows(one-or-more), right-rows(zero-or-more)
@@ -321,12 +321,12 @@ project
       ├── stats: [rows=0.3]
       ├── fd: ()-->(4), (1)-->(2,3)
       ├── inner-join (inverted json_multi_col@sij_idx [as=t1])
-      │    ├── columns: t2.j:9(jsonb) "inverted_join_const_col_@3":12(string!null) "inverted_join_const_col_@4":13(int!null) t1.k:14(int!null) s:16(string!null) i:17(int!null)
+      │    ├── columns: t2.j:9(jsonb) t1.k:14(int!null) s:16(string!null) i:17(int!null)
       │    ├── prefix key columns: [12 13] = [16 17]
       │    ├── inverted-expr
       │    │    └── t1.j:15 @> t2.j:9 [type=bool]
       │    ├── stats: [rows=0.3, distinct(12)=0.3, null(12)=0, distinct(13)=0.3, null(13)=0, distinct(14)=0.299956504, null(14)=0, distinct(16)=0.3, null(16)=0, distinct(17)=0.3, null(17)=0, distinct(16,17)=0.3, null(16,17)=0]
-      │    ├── fd: ()-->(13,17), (14)-->(16), (12)==(16), (16)==(12), (13)==(17), (17)==(13)
+      │    ├── fd: ()-->(17), (14)-->(16)
       │    ├── project
       │    │    ├── columns: "inverted_join_const_col_@4":13(int!null) t2.j:9(jsonb) "inverted_join_const_col_@3":12(string!null)
       │    │    ├── stats: [rows=30, distinct(12)=3, null(12)=0, distinct(13)=1, null(13)=0]

--- a/pkg/sql/opt/xform/testdata/rules/join
+++ b/pkg/sql/opt/xform/testdata/rules/join
@@ -5781,12 +5781,12 @@ project
       ├── immutable
       ├── fd: ()-->(14), (1)-->(10)
       ├── inner-join (inverted nyc_neighborhoods@nyc_neighborhoods_geo_idx [as=n])
-      │    ├── columns: c.gid:1!null c.geom:10 "inverted_join_const_col_@14":20!null n.gid:21!null n.boroname:22!null
+      │    ├── columns: c.gid:1!null c.geom:10 n.gid:21!null n.boroname:22!null
       │    ├── prefix key columns: [20] = [22]
       │    ├── inverted-expr
       │    │    └── st_covers(c.geom:10, n.geom:24)
       │    ├── key: (1,21)
-      │    ├── fd: ()-->(20,22), (1)-->(10), (20)==(22), (22)==(20)
+      │    ├── fd: ()-->(22), (1)-->(10)
       │    ├── project
       │    │    ├── columns: "inverted_join_const_col_@14":20!null c.gid:1!null c.geom:10
       │    │    ├── key: (1)
@@ -5819,12 +5819,12 @@ project
       ├── immutable
       ├── fd: ()-->(6,9)
       ├── inner-join (inverted json_arr1@a_idx [as=t1])
-      │    ├── columns: t2.a:4 "inverted_join_const_col_@6":18!null t1.k:19!null
+      │    ├── columns: t2.a:4 t1.k:19!null
       │    ├── flags: force inverted join (into right side)
       │    ├── prefix key columns: [18] = [19]
       │    ├── inverted-expr
       │    │    └── t1.a:22 @> t2.a:4
-      │    ├── fd: ()-->(18,19), (18)==(19), (19)==(18)
+      │    ├── fd: ()-->(19)
       │    ├── project
       │    │    ├── columns: "inverted_join_const_col_@6":18!null t2.a:4
       │    │    ├── fd: ()-->(18)
@@ -5854,12 +5854,12 @@ project
       ├── immutable
       ├── fd: ()-->(6,8)
       ├── inner-join (inverted json_arr1@j_idx [as=t1])
-      │    ├── columns: t2.j:3 "inverted_join_const_col_@6":17!null t1.k:18!null
+      │    ├── columns: t2.j:3 t1.k:18!null
       │    ├── flags: force inverted join (into right side)
       │    ├── prefix key columns: [17] = [18]
       │    ├── inverted-expr
       │    │    └── t1.j:20 @> t2.j:3
-      │    ├── fd: ()-->(17,18), (17)==(18), (18)==(17)
+      │    ├── fd: ()-->(18)
       │    ├── project
       │    │    ├── columns: "inverted_join_const_col_@6":17!null t2.j:3
       │    │    ├── fd: ()-->(17)
@@ -5889,12 +5889,12 @@ project
       ├── immutable
       ├── fd: ()-->(6,8,9)
       ├── inner-join (inverted json_arr1@j_idx [as=t1])
-      │    ├── columns: t2.j:3 "inverted_join_const_col_@6":17!null t1.k:18!null
+      │    ├── columns: t2.j:3 t1.k:18!null
       │    ├── flags: force inverted join (into right side)
       │    ├── prefix key columns: [17] = [18]
       │    ├── inverted-expr
       │    │    └── t1.j:20 @> t2.j:3
-      │    ├── fd: ()-->(17,18), (17)==(18), (18)==(17)
+      │    ├── fd: ()-->(18)
       │    ├── project
       │    │    ├── columns: "inverted_join_const_col_@6":17!null t2.j:3
       │    │    ├── fd: ()-->(17)
@@ -5924,12 +5924,11 @@ project
       ├── immutable
       ├── fd: (6)-->(8)
       ├── inner-join (inverted json_arr1@j_idx [as=t1])
-      │    ├── columns: t2.j:3 "inverted_join_const_col_@6":17!null t1.k:18!null
+      │    ├── columns: t2.j:3 t1.k:18!null
       │    ├── flags: force inverted join (into right side)
       │    ├── prefix key columns: [17] = [18]
       │    ├── inverted-expr
       │    │    └── t1.j:20 @> t2.j:3
-      │    ├── fd: (17)==(18), (18)==(17)
       │    ├── inner-join (cross)
       │    │    ├── columns: t2.j:3 "inverted_join_const_col_@6":17!null
       │    │    ├── multiplicity: left-rows(one-or-more), right-rows(zero-or-more)
@@ -5997,12 +5996,12 @@ project
       ├── immutable
       ├── fd: (6)-->(8)
       ├── inner-join (inverted json_check@secondary [as=t1])
-      │    ├── columns: t2.j:3 "inverted_join_const_col_@7":11!null t1.k:12!null i:13!null
+      │    ├── columns: t2.j:3 t1.k:12!null i:13!null
       │    ├── flags: force inverted join (into right side)
       │    ├── prefix key columns: [11] = [13]
       │    ├── inverted-expr
       │    │    └── t1.j:14 @> t2.j:3
-      │    ├── fd: (12)-->(13), (11)==(13), (13)==(11)
+      │    ├── fd: (12)-->(13)
       │    ├── inner-join (cross)
       │    │    ├── columns: t2.j:3 "inverted_join_const_col_@7":11!null
       │    │    ├── multiplicity: left-rows(one-or-more), right-rows(zero-or-more)
@@ -6036,12 +6035,12 @@ project
       ├── immutable
       ├── fd: ()-->(7), (6)-->(9)
       ├── inner-join (inverted json_comp@bj_idx [as=t1])
-      │    ├── columns: t2.j:3 "inverted_join_const_col_@8":12!null t1.k:13!null b:15!null
+      │    ├── columns: t2.j:3 t1.k:13!null b:15!null
       │    ├── flags: force inverted join (into right side)
       │    ├── prefix key columns: [12] = [15]
       │    ├── inverted-expr
       │    │    └── t1.j:16 @> t2.j:3
-      │    ├── fd: ()-->(12,15), (12)==(15), (15)==(12)
+      │    ├── fd: ()-->(15)
       │    ├── project
       │    │    ├── columns: "inverted_join_const_col_@8":12!null t2.j:3
       │    │    ├── fd: ()-->(12)
@@ -6081,12 +6080,12 @@ project
       ├── immutable
       ├── fd: ()-->(7), (6)-->(9)
       ├── inner-join (inverted json_comp@bj_partial_idx,partial [as=t1])
-      │    ├── columns: t2.j:3 "inverted_join_const_col_@8":13!null t1.k:14!null b:16!null
+      │    ├── columns: t2.j:3 t1.k:14!null b:16!null
       │    ├── flags: force inverted join (into right side)
       │    ├── prefix key columns: [13] = [16]
       │    ├── inverted-expr
       │    │    └── t1.j:17 @> t2.j:3
-      │    ├── fd: ()-->(13,16), (13)==(16), (16)==(13)
+      │    ├── fd: ()-->(16)
       │    ├── project
       │    │    ├── columns: "inverted_join_const_col_@8":13!null t2.j:3
       │    │    ├── fd: ()-->(13)
@@ -6153,12 +6152,12 @@ project
       ├── immutable
       ├── fd: (6)-->(7,8)
       ├── inner-join (inverted json_arr1@j_idx [as=t1])
-      │    ├── columns: t2.j:3 "inverted_join_const_col_@6":19!null "inverted_join_const_col_@7":20!null t1.k:21!null i:22!null
+      │    ├── columns: t2.j:3 t1.k:21!null i:22!null
       │    ├── flags: force inverted join (into right side)
       │    ├── prefix key columns: [19 20] = [21 22]
       │    ├── inverted-expr
       │    │    └── t1.j:23 @> t2.j:3
-      │    ├── fd: (21)-->(22), (19)==(21), (21)==(19), (20)==(22), (22)==(20)
+      │    ├── fd: (21)-->(22)
       │    ├── inner-join (cross)
       │    │    ├── columns: t2.j:3 "inverted_join_const_col_@6":19!null "inverted_join_const_col_@7":20!null
       │    │    ├── multiplicity: left-rows(one-or-more), right-rows(zero-or-more)
@@ -6203,12 +6202,12 @@ project
       ├── key: (6)
       ├── fd: (1)-->(3), (6)-->(7,8), (1)==(6), (6)==(1)
       ├── inner-join (inverted json_arr1@j_idx [as=t1])
-      │    ├── columns: t2.k:1!null t2.j:3 "inverted_join_const_col_@7":18!null t1.k:19!null i:20!null
+      │    ├── columns: t2.k:1!null t2.j:3 t1.k:19!null i:20!null
       │    ├── flags: force inverted join (into right side)
       │    ├── prefix key columns: [1 18] = [19 20]
       │    ├── inverted-expr
       │    │    └── t1.j:21 @> t2.j:3
-      │    ├── fd: (1)-->(3), (19)-->(20), (1)==(19), (19)==(1), (18)==(20), (20)==(18)
+      │    ├── fd: (1)-->(3), (19)-->(20), (1)==(19), (19)==(1)
       │    ├── inner-join (cross)
       │    │    ├── columns: t2.k:1!null t2.j:3 "inverted_join_const_col_@7":18!null
       │    │    ├── multiplicity: left-rows(one-or-more), right-rows(zero-or-more)
@@ -6322,11 +6321,11 @@ semi-join (lookup json_arr1 [as=t1])
  ├── key: (1)
  ├── fd: (1)-->(2-4)
  ├── inner-join (inverted json_arr1@j_idx [as=t1])
- │    ├── columns: t2.k:1!null l:2 t2.j:3 t2.a:4 "inverted_join_const_col_@7":19!null t1.k:20!null i:21!null continuation:33
+ │    ├── columns: t2.k:1!null l:2 t2.j:3 t2.a:4 t1.k:20!null i:21!null continuation:33
  │    ├── prefix key columns: [19] = [21]
  │    ├── inverted-expr
  │    │    └── t1.j:22 @> t2.j:3
- │    ├── fd: (1)-->(2-4), (20)-->(21,33), (19)==(21), (21)==(19)
+ │    ├── fd: (1)-->(2-4), (20)-->(21,33)
  │    ├── inner-join (cross)
  │    │    ├── columns: t2.k:1!null l:2 t2.j:3 t2.a:4 "inverted_join_const_col_@7":19!null
  │    │    ├── multiplicity: left-rows(one-or-more), right-rows(zero-or-more)

